accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,277 @@
+This RFC describes how Uyuni/SUSE Manager can prevent the user to perform these kinds of actions on the CaaS Platform nodes.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+## Terminology
+
+In this RFC, we will use:
+
+* CaaS Platform node: any node that comprises the cluster, either a control plane node or a worker node
+* Bootstrap/registration in the Uyuni/SUSE Manager context: registering a node to Uyuni/SUSE Manager via [the documented methods](https://opensource.suse.com/doc-susemanager/suse-manager/client-configuration/registration-overview.html).
+At the time of writing, registering a CaaS Platform node to Uyuni/SUSE Manager has to be manually performed by the user with:
+  * Bootstrap script
+  * UI: Systems > Bootstrapping
+  * `bootstrap` XMLRPC call
+  * Not usual but still possible and hackier: mass bootstrap via salt-ssh and a user-generated roster file
+
+## Assumptions
+
+We assume that:
+* The user is registering the CaaS Platform nodes as Salt clients
+* The registration of every CaaS Platform node to Uyuni/SUSE Manager is already completed by the user using the aforementioned methods
+* The user has already created an activation key and associated it with the onboarding. The CaaS Platform nodes are registered correctly and with the proper CaaS Platform channel(s) assigned.
+
+NOTE: It is outside of this RFC to discuss automatic registration methods of the CaaS Platform nodes.
+
+## Technical description of the problem
+
+### The forbidden actions
+
+[`skuba-update`](https://documentation.suse.com/suse-caasp/4/single-html/caasp-admin/#_base_os_updates) systemd timer will take care of [patching](https://github.com/SUSE/skuba/blob/master/skuba-update/skuba_update/skuba_update.py#L301-L305) all installed packages locally to every node of the cluster and optionally reboot the node if `zypper` signals that a reboot is required.
+
+Uyuni/SUSE Manager should not offer the possibility to reboot a registered CaaS Platform node.
+
+In general, patch and package upgrade/removal are safe operations for a CaaS Platform cluster: on the next trigger of `skuba-update`, if there are packages that required a reboot, `skuba-update` will orchestrate the reboot.
+
+Package upgrade/removal and patch apply is not a problem if issued from Uyuni/SUSE Manager, unless the involved packages of the action are one of the following:
+
+- `kubernetes-kubeadm`
+- `kubernetes-kubelet`
+- `kubernetes-client`
+- `cri-o`
+- `cni-plugins`
```

Comment:
What about dependencies of these packages, do they have any?

What if I remove one such dependency, will it remove the blacklist package as well?

Reply:
Good catch. It is right now unclear (on the CaaSP side) whether to consider forbidden the dependencies of the forbidden packages as well.
It is also difficult to construct the whole dependency tree of the dependencies with `zypper` (it will be available on 1.14.33) but right now it is a matter of iterating over `zypper info` and `zypper search`.